src/include.am

```diff
@@ -22,6 +22,7 @@ EXTRA_DIST += src/pk.c
 EXTRA_DIST += src/ssl_asn1.c
 EXTRA_DIST += src/ssl_bn.c
 EXTRA_DIST += src/ssl_misc.c
+EXTRA_DIST += src/version.c
```

Comment:
Adding a new files requires updating all the other build platforms like CMake, Visual Studio projects, etc. Please don't do this action unless a new file is absolutely required.

Reply:
This shouldn't be a reason not to separate out functionality into smaller files. Our files are too big as it is. Instead, maybe update the other build platforms.

Reply:
> This is very ESP centric

Indeed this is very ESP centric at the moment, as I've been doing mostly Espressif development. My plan is to have an infrastructure in place to allow others to add their own platform-specific details as needed & desired. When doing so, and only as needed, the respective build & make platforms could be updated. 

Although Espressif-specific now, I do see there being significant value in having a template of common system information for all platforms and all products. That would seem to be a valuable resource for customers to provide when asking for support.

>  It should be in the ESP specific code or even the ESP port code

I see your point here on the Espressif-specific details. I envisioned all of the "hardware / configuration overview" details being in a single file. As hardware typically has a version as well, that's another reason I was thinking of the `version.c` as a home. for this information.  

If instead we had something way down in the `./port/expressif`, what structure would you use? Perhaps a hardware specific `ShowHardwareConfiguration()` that is implemented for each platform, then called as appropriate from a single function elsewhere?

> don't [add version.c] unless a new file is absolutely required

It was quite a learning curve to tease that file into the builds, so I definitely respect that comment. Still, iIntuitively for me, it seems to make sense to have this information in the `version.c` file. That seems to be a universal file & the most global across all wolfSSL libraries. 

> Why not extend the existing `HAVE_WC_INTROSPECTION`?

This is an excellent question.  Although similar in concept, some of the information is mutually exclusive from introspection. @douzzer pointed out that: "_`HAVE_WC_INTROSPECTION` requires that the configuration and build time artifacts, particularly the date/time of build and git parameters, be excluded from the build_". For example: some of the information such as git hash & date may be updated, but the binary build itself should not actually change.

Given this, if you still feel strongly that I should not add the `version.c`, I'm happy to implement this functionality elsewhere.  Any preference? Perhaps `wolfcrypt/logging.c` would be a appropriate? I certainly agree with @bandi13 about the big files. I'm in favor of more specific, smaller files when possible.